jax/test_util.py

```diff
@@ -661,6 +661,9 @@ def rand(shape, dtype):
 
 def rand_int(rng, low=0, high=None):
   def fn(shape, dtype):
+    nonlocal high
+    if low == 0 and high is None:
+      high = np.iinfo(dtype).max
```

Comment:
We have places in the test suite where `rand_int` is used with floating dtypes. They always explicitly set `high` as far as I can tell, but we should account for the possibiility of non-integer dtypes here.

Reply:
I changed this code to report an error if `high` is not provided (because I couldn't think of a sensible choice of upper bound otherwise)